examples/models/llava_encoder/main.py

```diff
@@ -0,0 +1,201 @@
+from torch import nn
+
+from llava.model.builder import load_pretrained_model
+from llava.mm_utils import get_model_name_from_path
+from llava.eval.run_llava import eval_model, load_images, process_images
+# from executorch.exir import EdgeProgramManager, ExecutorchProgramManager, to_edge
+
+from llava.model.multimodal_encoder.builder import build_vision_tower
+
+# model_path = "liuhaotian/llava-v1.6-vicuna-7b"
+# only this one works
+model_path = "liuhaotian/llava-v1.5-7b"
+
+# not full model
+# model_path = "liuhaotian/llava-pretrain-llama-2-7b-chat"
+# torch.set_default_dtype(torch.float32)
+
+# class ImageEncoder(nn.Module):
+#     def __init__(self, model_path):
+#         super().__init__()
+#         tokenizer, self.model_, image_processor, context_len = load_pretrained_model(
+#             model_path=model_path,
+#             model_base=None,
+#             model_name=get_model_name_from_path(model_path)
+#         )
+#
+#     def forward(self, images_tensor):
+#
+# image_file = "https://llava-vl.github.io/static/images/view.jpg"
+# images = load_images([image_file])
+# image_sizes = [x.size for x in images]
+# images_tensor = process_images(
+#     images,
+#     image_processor,
+#     model.config
+# ).to(model.device, dtype=torch.float32)
+#
+# image_features = model.get_model().get_vision_tower()(images_tensor).float()
+
+
+prompt = "What are the things I should be cautious about when I visit here?"
+image_files = ['/Users/myuan/Downloads/pyturkeys.jpg']  # IMG_3997
+
+args = type('Args', (), {
+    "model_path": model_path,
+    "model_base": None,
+    "model_name": get_model_name_from_path(model_path),
+    "query": prompt,
+    "conv_mode": None,
+    "image_file": image_files[0],
+    "sep": ",",
+    "temperature": 0,
+    "top_p": None,
+    "num_beams": 1,
+    "max_new_tokens": 512
+})()
+
+model_name = get_model_name_from_path(args.model_path)
+tokenizer, model, image_processor, context_len = load_pretrained_model(
+    args.model_path, args.model_base, model_name, device_map="cpu", device="cpu"
+)
+
+# # uncomment this line for end to end eager mode run
+# eval_model(args)
+
+import torch
+from torch import nn
+import matplotlib.pyplot as plt
+
+plt.rcParams["savefig.bbox"] = 'tight'
+# import matplotlib.pyplot as plt
+
+from torchvision.transforms import v2
+from torchvision.transforms._functional_tensor import resize
+from torchvision.io import read_image
+import torchvision
+from helpers import plot
+
+from torch.export import export
+from torch._export import capture_pre_autograd_graph
+from torchvision.transforms.v2 import functional as F
+from torch.export import Dim
+from dataclasses import dataclass
+
+imagr = torchvision.io.read_image(image_files[0])
+imagt = imagr.to(dtype=torch.float16)
+print(imagt.shape)
+
+
+# plot([imagt])
+
+@dataclass
+class PreprocessConfig:
+    crop_size: dict
+    image_mean: list[float]
+    image_std: list[float]
+    rescale_factor: float
+
+
+class Preprocess(nn.Module):
```

Comment:
:+1: I'm waiting for this PR ! :>

And also im waiting for LLaVA full code.
Currently, i'm trying to do steps you wrote in README.md (llama connect) but not easy to do. 
I think current llama code in examples is incompatible with `liuhaotian/llava-v1.5-7b` one.

Reply:
Yeah please stay tuned we are looking into how they connect together.